service/controller/resource/clusterconditions/conditionproviderinfrastructureready.go

```diff
@@ -38,5 +38,8 @@ func (r *Resource) ensureProviderInfrastructureReadyCondition(ctx context.Contex
 		capi.ConditionSeverityWarning,
 		"AzureCluster Ready condition is not yet set, check again in few minutes")
 	capiconditions.SetMirror(cluster, aeconditions.ProviderInfrastructureReadyCondition, azureCluster, fallbackToFalse)
+
+	cluster.Status.InfrastructureReady = capiconditions.IsTrue(azureCluster, capi.ReadyCondition)
```

Comment:
Maybe here use Cluster ProviderInfrastructureReady, instead of AzureCluster Ready, the effect us the same, just maybe more explicit about relation between Cluster Status.InfrastructureReady and Cluster ProviderInfrastructureReady condition.